### ISSUE TITLE
Forgery::Basic.encrypt not generating unique values

### ISSUE THREAD

**Issue by mbbertino** [sevenwire/forgery#85]
Gem version `0.7.0`

I've got a model with an db index that has db unique constraint
``` ruby 
create_table "thing", force: :cascade do |t|
  t.string "name"
  t.string "unique_ref"
  t.index ["unique_ref"], name: "index_thing_on_unique_ref", unique: true
end
```

I've set up my factory with the following. (made sure I didn't make the mistake mentioned https://github.com/sevenwire/forgery/issues/43)
``` ruby
FactoryBot.define do
  factory :thing do
    unique_ref { Forgery::Basic.encrypt }
  end
end
```

When I go to run my spec
``` ruby
create(:thing, name: 'one')
create(:thing, name: 'two')
```

I receive the following error
```
ActiveRecord::RecordNotUnique:
  PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint "index_things_on_unique_ref"
  DETAIL:  Key (unique_ref)=(25357e2fe475921386b7bf09b9804d9e357ad287) already exists.
  : INSERT INTO "things" ("unique_ref", "name", "created_at", "updated_at") VALUES ($1, $2, $3, $4) RETURNING "id"
```

Switching from [`encrypt`](https://github.com/sevenwire/forgery/blob/80a10dfefcf452e2fadab300c54e31fa9ea276d9/lib/forgery/forgery/basic.rb#L53) to [`password`](https://github.com/sevenwire/forgery/blob/80a10dfefcf452e2fadab300c54e31fa9ea276d9/lib/forgery/forgery/basic.rb#L33) things seem to work as expected and generate unique values.
``` ruby
FactoryBot.define do
  factory :thing do
    unique_ref { Forgery::Basic.password }
  end
end
```

**Comment by brandonarbini** [comment]
Encrypt takes two optional arguments: `password` and `salt`. If you don't pass them it uses the default. In this case I'm guessing that `.to_s` on `Time` is only resolving to a second instead of millisecond and you're hitting a condition where the salt is the same. As a workaround you can pass a unique salt. However, we should also consider using `.to_f` to get some milliseconds in there.

```
def self.encrypt(password="password", salt=::Time.now.to_s)
    Digest::SHA1.hexdigest("--#{salt}--#{password}--")
  end
```

**Comment by mbbertino** [comment]
@brandonarbini that explanation would make total sense. 

I'm wondering if another path forward would be to utilize `rand` for the salt. Then you'd remove the need to `.to_s` or `.to_f` altogether. It seems like there isn't a need for it to be `Time` related unless I'm misunderstanding. 

Maybe something like: 

``` ruby 
def self.encrypt(password="password", salt=rand)
    Digest::SHA1.hexdigest("--#{salt}--#{password}--")
end
```

**Comment by brandonarbini** [comment]
That looks good to me.

**Comment by brandonarbini** [comment]
Released in v0.8.1. 👍
